src/backend/distributed/commands/utility_hook.c

```diff
@@ -124,6 +155,10 @@ static bool IsDropSchemaOrDB(Node *parsetree);
 static bool ShouldCheckUndistributeCitusLocalTables(void);
 static void RunPreprocessMainDBCommand(Node *parsetree, const char *queryString);
 static void RunPostprocessMainDBCommand(Node *parsetree);
+static bool IsStatementSupportedInNonMainDb(Node *parsetree);
+static bool StatementRequiresMarkDistributedFromNonMainDb(Node *parsetree);
+static ObjectInfo GetObjectInfo(Node *parsetree);
+static void MarkObjectDistributedInNonMainDb(Node *parsetree, ObjectInfo objectInfo);
```

Comment:
Thinking loudly; Probably we would want to move those functions into a new file but maybe in a separate PR ..

Reply:
What will be the function of this new file. I prefer moving it in another PR since I have chained 2 more PRs. Having a high refactoring, creates a high conflict so merge payload for me

Reply:
Yeah I'm also okay with doing this another PR as I mentioned.

> What will be the function of this new file. 

The purpose is to avoid making this file even bigger. This is also what we've done for main db DDLs by adding distribute_object_ops.c.